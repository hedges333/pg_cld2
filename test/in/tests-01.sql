SELECT "testing 1 2 3 mic check" AS mic_check;
\c regression;
\echo creating schema
CREATE SCHEMA test_schema;
\echo setting search_path
SET search_path TO test_schema;
\echo creating extension
CREATE EXTENSION pg_cld2 SCHEMA test_schema;
