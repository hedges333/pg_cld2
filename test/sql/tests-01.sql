SELECT 'testing 1 2 3 mic check' AS mic_check;

CREATE SCHEMA test_schema;

SET search_path TO test_schema;

CREATE EXTENSION pg_cld2 SCHEMA test_schema;

SELECT 'again testing 1 2 3 mic check' AS mic_check;
