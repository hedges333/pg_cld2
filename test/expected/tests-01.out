\x on
-- \set ECHO queries
\set sample_csv `pwd`/test/sql/sample_data.csv
CREATE SCHEMA test_schema;
SET search_path TO test_schema;
CREATE EXTENSION pg_cld2 SCHEMA test_schema;
CREATE TABLE sample_text (
    id                  INTEGER NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
    text_to_analyze         TEXT    NOT NULL,
    is_plain_text           BOOLEAN,
    content_language_hint   TEXT,
    tld_hint                TEXT,
    cld2_language_hint      TEXT,
    best_effort             BOOLEAN,
    text_encoding           TEXT,
    tsconfig_language_hint  TEXT,
    locale_lang_hint        TEXT
);
COPY sample_text(
    text_to_analyze,
    is_plain_text,
    content_language_hint,
    tld_hint,
    cld2_language_hint,
    best_effort,
    text_encoding,
    tsconfig_language_hint,
    locale_lang_hint
)
FROM :'sample_csv'
WITH (FORMAT csv, HEADER true);
SELECT('check 1 2 3') AS mic_check;
-[ RECORD 1 ]----------
mic_check | check 1 2 3

CREATE OR REPLACE FUNCTION process_sample_texts()
RETURNS TABLE (
    sample_text_id                          INTEGER,
    sample_text_text_to_analyze_substr      TEXT,
    sample_text_is_plain_text               BOOLEAN,
    sample_text_content_language_hint       TEXT,
    sample_text_tld_hint                    TEXT,
    sample_text_cld2_language_hint          TEXT,
    sample_text_best_effort                 BOOLEAN,
    sample_text_text_encoding               TEXT,
    sample_text_tsconfig_language_hint      TEXT,
    sample_text_locale_lang_hint            TEXT,
    pg_cld2_input_bytes                     INTEGER,
    pg_cld2_text_bytes                      INTEGER,
    pg_cld2_is_reliable                     BOOLEAN,
    pg_cld2_valid_prefix_bytes              INTEGER,
    pg_cld2_is_valid_utf8                   BOOLEAN,
    pg_cld2_mll_cld2_name                   TEXT,
    pg_cld2_mll_language_cname              TEXT,
    pg_cld2_mll_language_code               TEXT,
    pg_cld2_mll_primary_script_name         TEXT,
    pg_cld2_mll_primary_script_code         TEXT,
    pg_cld2_mll_script_names                TEXT,
    pg_cld2_mll_script_codes                TEXT,
    pg_cld2_mll_ts_name                     TEXT,
    pg_cld2_language_1_cld2_name            TEXT,
    pg_cld2_language_1_language_cname       TEXT,
    pg_cld2_language_1_language_code        TEXT,
    pg_cld2_language_1_primary_script_name  TEXT,
    pg_cld2_language_1_primary_script_code  TEXT,
    pg_cld2_language_1_script_names         TEXT,
    pg_cld2_language_1_script_codes         TEXT,
    pg_cld2_language_1_percent              INTEGER,
    pg_cld2_language_1_normalized_score     DOUBLE PRECISION,
    pg_cld2_language_1_ts_name              TEXT,
    pg_cld2_language_2_cld2_name            TEXT,
    pg_cld2_language_2_language_cname       TEXT,
    pg_cld2_language_2_language_code        TEXT,
    pg_cld2_language_2_primary_script_name  TEXT,
    pg_cld2_language_2_primary_script_code  TEXT,
    pg_cld2_language_2_script_names         TEXT,
    pg_cld2_language_2_script_codes         TEXT,
    pg_cld2_language_2_percent              INTEGER,
    pg_cld2_language_2_normalized_score     DOUBLE PRECISION,
    pg_cld2_language_2_ts_name              TEXT,
    pg_cld2_language_3_cld2_name            TEXT,
    pg_cld2_language_3_language_cname       TEXT,
    pg_cld2_language_3_language_code        TEXT,
    pg_cld2_language_3_primary_script_name  TEXT,
    pg_cld2_language_3_primary_script_code  TEXT,
    pg_cld2_language_3_script_names         TEXT,
    pg_cld2_language_3_script_codes         TEXT,
    pg_cld2_language_3_percent              INTEGER,
    pg_cld2_language_3_normalized_score     DOUBLE PRECISION,
    pg_cld2_language_3_ts_name              TEXT
) AS $$
BEGIN
    RETURN QUERY
    SELECT
        sample_text_rec.id,
        SUBSTRING(sample_text_rec.text_to_analyze FOR 32),
        sample_text_rec.is_plain_text,
        sample_text_rec.content_language_hint,
        sample_text_rec.tld_hint,
        sample_text_rec.cld2_language_hint,
        sample_text_rec.best_effort,
        sample_text_rec.text_encoding,
        sample_text_rec.tsconfig_language_hint,
        sample_text_rec.locale_lang_hint,
        result.input_bytes,
        result.text_bytes,
        result.is_reliable,
        result.valid_prefix_bytes,
        result.is_valid_utf8,
        result.mll_cld2_name,
        result.mll_language_cname,
        result.mll_language_code,
        result.mll_primary_script_name,
        result.mll_primary_script_code,
        result.mll_script_names,
        result.mll_script_codes,
        result.mll_ts_name,
        result.language_1_cld2_name,
        result.language_1_language_cname,
        result.language_1_language_code,
        result.language_1_primary_script_name,
        result.language_1_primary_script_code,
        result.language_1_script_names,
        result.language_1_script_codes,
        result.language_1_percent,
        result.language_1_normalized_score,
        result.language_1_ts_name,
        result.language_2_cld2_name,
        result.language_2_language_cname,
        result.language_2_language_code,
        result.language_2_primary_script_name,
        result.language_2_primary_script_code,
        result.language_2_script_names,
        result.language_2_script_codes,
        result.language_2_percent,
        result.language_2_normalized_score,
        result.language_2_ts_name,
        result.language_3_cld2_name,
        result.language_3_language_cname,
        result.language_3_language_code,
        result.language_3_primary_script_name,
        result.language_3_primary_script_code,
        result.language_3_script_names,
        result.language_3_script_codes,
        result.language_3_percent,
        result.language_3_normalized_score,
        result.language_3_ts_name
    FROM sample_text sample_text_rec,
    LATERAL (
        SELECT * FROM pg_cld2_detect_language(
            sample_text_rec.text_to_analyze,
            sample_text_rec.is_plain_text,
            sample_text_rec.content_language_hint,
            sample_text_rec.tld_hint,
            sample_text_rec.cld2_language_hint,
            sample_text_rec.best_effort,
            sample_text_rec.text_encoding,
            sample_text_rec.tsconfig_language_hint,
            sample_text_rec.locale_lang_hint
        )
    ) AS result;
END;
$$ LANGUAGE plpgsql;
SELECT * from process_sample_texts();
-[ RECORD 1 ]--------------------------+---------------------------------
sample_text_id                         | 1
sample_text_text_to_analyze_substr     | Mistah Kurtz - he dead.         +
                                       |                                 +
                                       | A penny
sample_text_is_plain_text              | t
sample_text_content_language_hint      | 
sample_text_tld_hint                   | 
sample_text_cld2_language_hint         | 
sample_text_best_effort                | 
sample_text_text_encoding              | 
sample_text_tsconfig_language_hint     | 
sample_text_locale_lang_hint           | 
pg_cld2_input_bytes                    | 2495
pg_cld2_text_bytes                     | 2354
pg_cld2_is_reliable                    | t
pg_cld2_valid_prefix_bytes             | 2495
pg_cld2_is_valid_utf8                  | f
pg_cld2_mll_cld2_name                  | ENGLISH
pg_cld2_mll_language_cname             | ENGLISH
pg_cld2_mll_language_code              | en
pg_cld2_mll_primary_script_name        | Latin
pg_cld2_mll_primary_script_code        | Latn
pg_cld2_mll_script_names               | Latin
pg_cld2_mll_script_codes               | Latn
pg_cld2_mll_ts_name                    | english
pg_cld2_language_1_cld2_name           | ENGLISH
pg_cld2_language_1_language_cname      | ENGLISH
pg_cld2_language_1_language_code       | en
pg_cld2_language_1_primary_script_name | Latin
pg_cld2_language_1_primary_script_code | Latn
pg_cld2_language_1_script_names        | Latin
pg_cld2_language_1_script_codes        | Latn
pg_cld2_language_1_percent             | 99
pg_cld2_language_1_normalized_score    | 6.9e-321
pg_cld2_language_1_ts_name             | english
pg_cld2_language_2_cld2_name           | Unknown
pg_cld2_language_2_language_cname      | UNKNOWN_LANGUAGE
pg_cld2_language_2_language_code       | un
pg_cld2_language_2_primary_script_name | Latin
pg_cld2_language_2_primary_script_code | Latn
pg_cld2_language_2_script_names        | Latin
pg_cld2_language_2_script_codes        | Latn
pg_cld2_language_2_percent             | 0
pg_cld2_language_2_normalized_score    | 0
pg_cld2_language_2_ts_name             | simple
pg_cld2_language_3_cld2_name           | Unknown
pg_cld2_language_3_language_cname      | UNKNOWN_LANGUAGE
pg_cld2_language_3_language_code       | un
pg_cld2_language_3_primary_script_name | Latin
pg_cld2_language_3_primary_script_code | Latn
pg_cld2_language_3_script_names        | Latin
pg_cld2_language_3_script_codes        | Latn
pg_cld2_language_3_percent             | 0
pg_cld2_language_3_normalized_score    | 0
pg_cld2_language_3_ts_name             | simple
-[ RECORD 2 ]--------------------------+---------------------------------
sample_text_id                         | 2
sample_text_text_to_analyze_substr     | Four score and seven years ago o
sample_text_is_plain_text              | t
sample_text_content_language_hint      | 
sample_text_tld_hint                   | 
sample_text_cld2_language_hint         | 
sample_text_best_effort                | 
sample_text_text_encoding              | 
sample_text_tsconfig_language_hint     | 
sample_text_locale_lang_hint           | 
pg_cld2_input_bytes                    | 1449
pg_cld2_text_bytes                     | 1418
pg_cld2_is_reliable                    | t
pg_cld2_valid_prefix_bytes             | 1449
pg_cld2_is_valid_utf8                  | f
pg_cld2_mll_cld2_name                  | ENGLISH
pg_cld2_mll_language_cname             | ENGLISH
pg_cld2_mll_language_code              | en
pg_cld2_mll_primary_script_name        | Latin
pg_cld2_mll_primary_script_code        | Latn
pg_cld2_mll_script_names               | Latin
pg_cld2_mll_script_codes               | Latn
pg_cld2_mll_ts_name                    | english
pg_cld2_language_1_cld2_name           | ENGLISH
pg_cld2_language_1_language_cname      | ENGLISH
pg_cld2_language_1_language_code       | en
pg_cld2_language_1_primary_script_name | Latin
pg_cld2_language_1_primary_script_code | Latn
pg_cld2_language_1_script_names        | Latin
pg_cld2_language_1_script_codes        | Latn
pg_cld2_language_1_percent             | 99
pg_cld2_language_1_normalized_score    | 7.76e-321
pg_cld2_language_1_ts_name             | english
pg_cld2_language_2_cld2_name           | Unknown
pg_cld2_language_2_language_cname      | UNKNOWN_LANGUAGE
pg_cld2_language_2_language_code       | un
pg_cld2_language_2_primary_script_name | Latin
pg_cld2_language_2_primary_script_code | Latn
pg_cld2_language_2_script_names        | Latin
pg_cld2_language_2_script_codes        | Latn
pg_cld2_language_2_percent             | 0
pg_cld2_language_2_normalized_score    | 0
pg_cld2_language_2_ts_name             | simple
pg_cld2_language_3_cld2_name           | Unknown
pg_cld2_language_3_language_cname      | UNKNOWN_LANGUAGE
pg_cld2_language_3_language_code       | un
pg_cld2_language_3_primary_script_name | Latin
pg_cld2_language_3_primary_script_code | Latn
pg_cld2_language_3_script_names        | Latin
pg_cld2_language_3_script_codes        | Latn
pg_cld2_language_3_percent             | 0
pg_cld2_language_3_normalized_score    | 0
pg_cld2_language_3_ts_name             | simple
-[ RECORD 3 ]--------------------------+---------------------------------
sample_text_id                         | 3
sample_text_text_to_analyze_substr     | I                               +
                                       |                                 +
                                       | Marseille.--L'arrivée.          +
                                       |                                 +
                                       |                                 +
                                       | Le 2
sample_text_is_plain_text              | t
sample_text_content_language_hint      | 
sample_text_tld_hint                   | 
sample_text_cld2_language_hint         | 
sample_text_best_effort                | 
sample_text_text_encoding              | 
sample_text_tsconfig_language_hint     | 
sample_text_locale_lang_hint           | 
pg_cld2_input_bytes                    | 2362
pg_cld2_text_bytes                     | 2285
pg_cld2_is_reliable                    | t
pg_cld2_valid_prefix_bytes             | 2362
pg_cld2_is_valid_utf8                  | f
pg_cld2_mll_cld2_name                  | FRENCH
pg_cld2_mll_language_cname             | FRENCH
pg_cld2_mll_language_code              | fr
pg_cld2_mll_primary_script_name        | Latin
pg_cld2_mll_primary_script_code        | Latn
pg_cld2_mll_script_names               | Latin
pg_cld2_mll_script_codes               | Latn
pg_cld2_mll_ts_name                    | french
pg_cld2_language_1_cld2_name           | FRENCH
pg_cld2_language_1_language_cname      | FRENCH
pg_cld2_language_1_language_code       | fr
pg_cld2_language_1_primary_script_name | Latin
pg_cld2_language_1_primary_script_code | Latn
pg_cld2_language_1_script_names        | Latin
pg_cld2_language_1_script_codes        | Latn
pg_cld2_language_1_percent             | 99
pg_cld2_language_1_normalized_score    | 4.323e-321
pg_cld2_language_1_ts_name             | french
pg_cld2_language_2_cld2_name           | Unknown
pg_cld2_language_2_language_cname      | UNKNOWN_LANGUAGE
pg_cld2_language_2_language_code       | un
pg_cld2_language_2_primary_script_name | Latin
pg_cld2_language_2_primary_script_code | Latn
pg_cld2_language_2_script_names        | Latin
pg_cld2_language_2_script_codes        | Latn
pg_cld2_language_2_percent             | 0
pg_cld2_language_2_normalized_score    | 0
pg_cld2_language_2_ts_name             | simple
pg_cld2_language_3_cld2_name           | Unknown
pg_cld2_language_3_language_cname      | UNKNOWN_LANGUAGE
pg_cld2_language_3_language_code       | un
pg_cld2_language_3_primary_script_name | Latin
pg_cld2_language_3_primary_script_code | Latn
pg_cld2_language_3_script_names        | Latin
pg_cld2_language_3_script_codes        | Latn
pg_cld2_language_3_percent             | 0
pg_cld2_language_3_normalized_score    | 0
pg_cld2_language_3_ts_name             | simple
-[ RECORD 4 ]--------------------------+---------------------------------
sample_text_id                         | 4
sample_text_text_to_analyze_substr     | <?xml version="1.0" encoding="UT
sample_text_is_plain_text              | t
sample_text_content_language_hint      | 
sample_text_tld_hint                   | 
sample_text_cld2_language_hint         | 
sample_text_best_effort                | 
sample_text_text_encoding              | 
sample_text_tsconfig_language_hint     | 
sample_text_locale_lang_hint           | 
pg_cld2_input_bytes                    | 5257
pg_cld2_text_bytes                     | 4617
pg_cld2_is_reliable                    | t
pg_cld2_valid_prefix_bytes             | 5257
pg_cld2_is_valid_utf8                  | f
pg_cld2_mll_cld2_name                  | Japanese
pg_cld2_mll_language_cname             | JAPANESE
pg_cld2_mll_language_code              | ja
pg_cld2_mll_primary_script_name        | Hani
pg_cld2_mll_primary_script_code        | Hani
pg_cld2_mll_script_names               | Hani
pg_cld2_mll_script_codes               | Hani
pg_cld2_mll_ts_name                    | simple
pg_cld2_language_1_cld2_name           | Japanese
pg_cld2_language_1_language_cname      | JAPANESE
pg_cld2_language_1_language_code       | ja
pg_cld2_language_1_primary_script_name | Hani
pg_cld2_language_1_primary_script_code | Hani
pg_cld2_language_1_script_names        | Hani
pg_cld2_language_1_script_codes        | Hani
pg_cld2_language_1_percent             | 78
pg_cld2_language_1_normalized_score    | 1.4323e-320
pg_cld2_language_1_ts_name             | simple
pg_cld2_language_2_cld2_name           | ENGLISH
pg_cld2_language_2_language_cname      | ENGLISH
pg_cld2_language_2_language_code       | en
pg_cld2_language_2_primary_script_name | Latin
pg_cld2_language_2_primary_script_code | Latn
pg_cld2_language_2_script_names        | Latin
pg_cld2_language_2_script_codes        | Latn
pg_cld2_language_2_percent             | 17
pg_cld2_language_2_normalized_score    | 3.05e-321
pg_cld2_language_2_ts_name             | english
pg_cld2_language_3_cld2_name           | KINYARWANDA
pg_cld2_language_3_language_cname      | KINYARWANDA
pg_cld2_language_3_language_code       | rw
pg_cld2_language_3_primary_script_name | Latin
pg_cld2_language_3_primary_script_code | Latn
pg_cld2_language_3_script_names        | Latin
pg_cld2_language_3_script_codes        | Latn
pg_cld2_language_3_percent             | 3
pg_cld2_language_3_normalized_score    | 7.28e-321
pg_cld2_language_3_ts_name             | simple

